src/backend/joanie/core/models/quotes.py

```diff
@@ -138,6 +147,38 @@ class Meta:
     def __str__(self):
         return f"Quote for course {self.batch_order.relation.course.code}"
 
+    def clean(self):
+        """
+        When the object is created, add the a unique reference to it
+        """
+        if not self.reference:
+            year = timezone.now().year
+            with transaction.atomic():
+                # select_for_update() will lock row the queryset until the end of the transaction
+                last = (
+                    Quote.objects.filter(
+                        reference__startswith=f"{settings.JOANIE_QUOTE_REFERENCE_PREFIX}_{year}_"
```

Comment:
Maybe we could extract `f"{settings.JOANIE_QUOTE_REFERENCE_PREFIX}_{year}_"` to a variable ?